scripts/upgrade

```diff
@@ -14,28 +12,179 @@ source /usr/share/yunohost/helpers
 #=================================================
+ynh_print_info --message="Backing up the app before upgrading (may take a while)..."
+
 # Backup the current version of the app
 ynh_backup_before_upgrade
 ynh_clean_setup () {
-    # restore it if the upgrade fails
-    ynh_restore_upgradebackup
+	# restore it if the upgrade fails
+	ynh_restore_upgradebackup
 }
 # Exit if an error occurs during the execution of the script
 ynh_abort_if_errors
 
 #=================================================
 # STANDARD UPGRADE STEPS
 #=================================================
-# nothing yet
+# DOWNLOAD, CHECK AND UNPACK SOURCE
+#=================================================
+ynh_print_info --message="Upgrading source files..."
+
+# Remove old install
+ynh_secure_remove --file="$final_path"
+
+# Download, check integrity, uncompress and patch the source from app.src
+ynh_setup_source --dest_dir="$final_path"
+
+#=================================================
+# UPGRADE DEPENDENCIES
+#=================================================
+ynh_print_info --message="Upgrading dependencies..."
+
+ynh_install_app_dependencies $pkg_dependencies
+
+#=================================================
+# CREATE DEDICATED USER
+#=================================================
+ynh_print_info --message="Making sure dedicated system user exists..."
+
+# Create a dedicated user (if not existing)
+ynh_system_user_create --username=$app --home_dir=/home/yunohost.app/$app --use_shell
+
+#=================================================
+# SETUP SYSTEMD
+#=================================================
+ynh_print_info --message="Upgrading systemd configuration..."
+
+# Create a dedicated systemd config
+ynh_add_systemd_config
+
+#=================================================
+# COPY CONFIG FILE
+#=================================================
+
+if [ -e /home/yunohost.app/$app ]
+then
+echo "Nothing to do..."
+else
+mkdir -p /home/yunohost.app/$app/.minetest/
+mkdir /home/yunohost.app/$app/.minetest/worlds/
+fi
+
+cp -a ../conf/minetest.conf /home/yunohost.app/$app/.minetest/
+
+#=================================================
+# SPECIFIC UPGRADE
+#=================================================
+# Compile Minetest
+pushd /opt/yunohost/$app
+# Download Minetest Game
+ynh_setup_source --dest_dir=/opt/yunohost/$app/games/minetest_game --source_id=minetest_game
+
+cmake . -DRUN_IN_PLACE=TRUE -DENABLE_LUAJIT=TRUE -DBUILD_CLIENT=FALSE -DBUILD_SERVER=TRUE
+
+make -j$(nproc)
+
+popd
+
+### Verify the checksum of a file, stored by `ynh_store_file_checksum` in the install script.
+### And create a backup of this file if the checksum is different. So the file will be backed up if the admin had modified it.
+ynh_backup_if_checksum_is_different --file="$home_path/.minetest/minetest.conf"
+# Recalculate and store the checksum of the file for the next upgrade.
+ynh_store_file_checksum --file="$home_path/.minetest/minetest.conf"
```

Comment:
You should copy the default conf in the right place before:
`cp ../conf/minetest.conf /home/yunohost.app/$app/.minetest/`
then call all `ynh_replace_string` before `ynh_store_file_checksum`.

It should looks like that:

```
ynh_backup_if_checksum_is_different --file="$home_path/.minetest/minetest.conf"

cp ../conf/minetest.conf $home/.minetest/minetest.conf

ynh_replace_string --match_string="__APP__" --replace_string="$app" --target_file="$home_path/.minetest/minetest.conf"
ynh_replace_string --match_string="__PORT__" --replace_string="$port" --target_file="$home_path/.minetest/minetest.conf"
ynh_replace_string --match_string="__SERVERNAME__" --replace_string="$servername" --target_file="$home_path/.minetest/minetest.conf"
ynh_replace_string --match_string="__DOMAIN__" --replace_string="$domain" --target_file="$home_path/.minetest/minetest.conf"
ynh_replace_string --match_string="__ANNOUNCE__" --replace_string="$is_public" --target_file="$home_path/.minetest/minetest.conf"
ynh_replace_string --match_string="__PVP__" --replace_string="$pvp" --target_file="$home_path/.minetest/minetest.conf"
ynh_replace_string --match_string="__CREATIVE__" --replace_string="$creative" --target_file="$home_path/.minetest/minetest.conf"
ynh_replace_string --match_string="__DAMAGE__" --replace_string="$damage" --target_file="$home_path/.minetest/minetest.conf"

ynh_store_file_checksum --file="$home_path/.minetest/minetest.conf"
```

Reply:
Fixed :)